src/controllers/controller.bar.js

```diff
@@ -383,11 +388,16 @@ module.exports = function(Chart) {
 			var meta = me.getMeta();
 			var scale = me.getValueScale();
 			var datasets = chart.data.datasets;
-			var value = scale.getRightValue(datasets[datasetIndex].data[index]);
+			// float-bar support, if y arguments are array function will use top - bottom value to calculate bar height
+			var value = scale.parseValue(datasets[datasetIndex].data[index]);
 			var stacked = scale.options.stacked;
 			var stack = meta.stack;
+			// float-bar support, if y arguments are array function will use proper value as bar start point
 			var start = 0;
-			var i, imeta, ivalue, base, head, size;
+			var i, imeta, ivalue, base, head, size, yStackValue;
+			if (value.min !== value.max) {
+				start = value.val > 0 ? value.min : value.max;
```

Comment:
shouldn't this be `value.min` instead of `value.val`? `value.val` is the average of `min` and `max` and I'm not sure why that would be relevant

Reply:
it dodesnt matter here in fact what to check val, min or max, cause the main requirements both values in array should be positive or negative (logic shouldbe that bar doesnt croos the zero line). So, here we only check if value.val  is negative or posive and its true for min or max values, bacuse all those values should be positive or negative. I will change maybe only > to >=